xapian-letor/include/xapian-letor/ranker.h

```diff
@@ -193,7 +199,8 @@ class XAPIAN_VISIBILITY_DEFAULT Ranker : public Xapian::Internal::intrusive_base
   protected:
     /// Method to train the model. Overridden in ranker subclass.
     virtual void
-    train(const std::vector<Xapian::FeatureVector> & training_data) = 0;
+    train(const std::vector<std::vector<Xapian::FeatureVector>>&
+	  training_data) = 0;
```

Comment:
We're inside `namespace Xapian {` ... `}` here, so the explicit `Xapian::` qualification isn't necessary here.  And without there's no need to wrap the parameter list.